### PR TITLE
Add AAD support to job agents version 2023-05-01-preview

### PR DESCRIPTION
**Pull request by rayboyd30**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:310]: >JobAgentIdentity [](http://example.com/codeflow?start=5&length=16)

Please see if you can refactor this part to define it using definitions from common-types instead

e.g. with

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json)

or

[https://github.com/Azure/azure-rest-api-specs/blob/5dd1107d5f2be8d600325d795450e1d854fbe7e8/specification/common-types/resource-management/v5/types.json#L201](https://github.com/Azure/azure-rest-api-specs/blob/5dd1107d5f2be8d600325d795450e1d854fbe7e8/specification/common-types/resource-management/v5/types.json#L201)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:420]: >JobAgentUserAssignedIdentity [](http://example.com/codeflow?start=5&length=28)

Looks exactly like

[https://github.com/Azure/azure-rest-api-specs/blob/5dd1107d5f2be8d600325d795450e1d854fbe7e8/specification/common-types/resource-management/v5/managedidentity.json#L18C6-L18C26](https://github.com/Azure/azure-rest-api-specs/blob/5dd1107d5f2be8d600325d795450e1d854fbe7e8/specification/common-types/resource-management/v5/managedidentity.json#L18C6-L18C26)

**Comment by TimLovellSmith** [comment]
>Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)

Given the change involved and this being in public repo, I think this may actually require a new preview api-version.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (rayboyd30) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:310]: We would like to keep this consistent with other Microsoft.Sql APIs such as: https://github.com/Azure/azure-rest-api-specs/blob/def187e2e78d7173d8fdd7f77740dd9719e1dfbf/specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/Databases.json#L875C7-L875C7

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (rayboyd30) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:420]: We would like to keep this consistent with other Microsoft.Sql APIs such as: https://github.com/Azure/azure-rest-api-specs/blob/def187e2e78d7173d8fdd7f77740dd9719e1dfbf/specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/Databases.json#L875C7-L875C7

**Comment by rayboyd30** [comment]
> > Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
> 
> Given the change involved and this being in public repo, I think this may actually require a new preview api-version.

Is this not the new api version? 2023-05-01-preview. We need to make these changes in this version to support public preview refresh on 10/25/23

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (visingla-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:310]: Either define a common version of identity types under Microsoft.Sql RP or refer it from ARM Common Types. Don't define the same type for all different resource types separately in their swagger json files.

Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:310]: Just in case you want to define your own RP common types, I strongly recommend having them in your own 'common.json' or similar name json file, to help everyone remember that they're *shared* contracts (not to be broken), and also avoid accidental circular dependencies between JSON files, which some tools can't handle today.

Comment (rayboyd30) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:310]: aligning with SQL common type def needs extra efforts from SQL security team. We have to ask for an exception in current version. we will coordinate with SQL security team to fix it in next version. But for this version, please unblock.
Here is the work item to align common types: https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/2697929

Comment (visingla-ms) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:310]: Is it possible to set an ETA and link it to a Feature/user Story to track this task? Just want to make sure that the next person who makes changes for next API version, is able to address this.

Comment (rayboyd30) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:310]: The item has been added to SQL API committer task queue and will be fixed in the next API version.

**Review by raosuhas — APPROVED** [review]

**Pull request merged**
